examples/DataApi-NFT/template/modules/shared_data/src/main/scala/com/my/nft_example/shared_data/calculated_state/CalculatedState.scala

```diff
@@ -0,0 +1,49 @@
+package com.my.nft_example.shared_data.calculated_state
+
+import cats.effect.IO
+import com.my.nft_example.shared_data.types.Types.NFTUpdatesCalculatedState
+import eu.timepit.refined.types.numeric.NonNegLong
+import io.circe.syntax.EncoderOps
+import org.tessellation.schema.SnapshotOrdinal
+import org.tessellation.security.hash.Hash
+
+import java.security.MessageDigest
+
+object CalculatedState {
+
+  private var maybeNFTUpdatesCalculatedState: (SnapshotOrdinal, NFTUpdatesCalculatedState) = (
+    SnapshotOrdinal(NonNegLong(0L)),
+    NFTUpdatesCalculatedState(Map.empty)
+  )
+
+  def getCalculatedState: IO[(SnapshotOrdinal, NFTUpdatesCalculatedState)] = {
+    IO(maybeNFTUpdatesCalculatedState)
+  }
+
+  def setCalculatedState(snapshotOrdinal: SnapshotOrdinal, state: NFTUpdatesCalculatedState): IO[Boolean] = {
+    val currentCheckInCalculatedState = maybeNFTUpdatesCalculatedState._2
+    val updatedDevices = state.collections.foldLeft(currentCheckInCalculatedState.collections) {
+      case (acc, (address, value)) =>
+        acc.updated(address, value)
+    }
+
+    maybeNFTUpdatesCalculatedState = (
+      snapshotOrdinal,
+      NFTUpdatesCalculatedState(updatedDevices)
+    )
+
+    IO(true)
+  }
+
+  private def sha256Hash(input: String): String = {
```

Comment:
this is private but doesn't seem used here. If you do need it though, perhaps the sdk method in `import org.tessellation.security.hash.Hash` would be better suited?

Reply:
Good catch